Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/PolDiffILLReduction.py

```diff
@@ -0,0 +1,876 @@
+        for entry in mtd[ws]:
+            SetInstrumentParameter(entry, ParameterName="formula_eff", Value=correction_formula)
+            DetectorEfficiencyCorUser(InputWorkspace=entry, OutputWorkspace=entry,
+                                      IncidentEnergy=self._experimentProperties['initial_energy'].value)
+        return ws
+
+    def _frame_overlap_correction(self, ws):
+        pass
+
+    def _apply_polarization_corrections(self, ws, pol_eff_ws):
+        """Applies the polarisation correction based on the output from quartz reduction."""
+        fp = 1 # flipper efficiency, placeholder
+        nPolarisations = None
+        singleCorrectionPerPOL = False
+        if mtd[ws].getNumberOfEntries() != 2*mtd[pol_eff_ws].getNumberOfEntries():
+            singleCorrectionPerPOL = True
+            if self._method_data_structure == 'Uniaxial':
+                nPolarisations = 2
+            elif self._method_data_structure == 'XYZ':
+                nPolarisations = 3
+            elif self._method_data_structure == '10p':
+                nPolarisations = 5
+            if mtd[pol_eff_ws].getNumberOfEntries() != nPolarisations:
+                raise RuntimeError("Incompatible number of polarisations between quartz input and sample.")
+
+        for entry_no in range(mtd[ws].getNumberOfEntries()):
+            if entry_no % 2 != 0:
+                continue
+            polarisation_entry_no = int(entry_no/2)
+            if singleCorrectionPerPOL:
+                polarisation_entry_no = int(polarisation_entry_no % nPolarisations)
+            phi = mtd[pol_eff_ws].getItem(polarisation_entry_no).name()
+            intensity_0 = mtd[ws].getItem(entry_no).name()
+            intensity_1 = mtd[ws].getItem(entry_no+1).name()
+            tmp_names = [intensity_0 + '_tmp', intensity_1 + '_tmp']
+
+            Divide(LHSWorkspace=((1.0-mtd[phi])*(1.0-fp) + fp*(1+mtd[phi]))*mtd[intensity_0]
+                   -(1.0-mtd[phi])*mtd[intensity_1],
+                   RHSWorkspace=2.0 * fp * mtd[phi],
+                   OutputWorkspace=tmp_names[0])
+            Divide(LHSWorkspace=(1+mtd[phi])*mtd[intensity_1]
+                   - ( (1+mtd[phi])*(1-fp) - fp*(1-mtd[phi]) )*mtd[intensity_0],
+                   RHSWorkspace=2.0 * fp * mtd[phi],
+                   OutputWorkspace=tmp_names[1])
+
+            RenameWorkspace(tmp_names[0], intensity_0)
+            RenameWorkspace(tmp_names[1], intensity_1)
+
+        return ws
+
+    def _read_experiment_properties(self, ws):
+        """Reads the user-provided dictionary that contains sample geometry (type, dimensions) and experimental conditions,
+         such as the beam size and calculates derived parameters."""
+        self._experimentProperties = self.getProperty('ExperimentPropertiesDictionary').value
+        if 'n_atoms' not in self._experimentProperties:
+            self._experimentProperties['n_atoms'] = float(self._experimentProperties['number_density'].value) \
```

Comment:
Why cast to float?

Reply:
This was meant to ensure float-division in case some of these parameters are understood as int in the dictionary.